bark/world/objects/agent.hpp

```diff
@@ -172,12 +172,30 @@ class Agent : public Object {
 
   void SetRoadCorridor(const RoadCorridorPtr road_corridor) {
     road_corridor_ = road_corridor;
+    road_corridor_road_ids_ = road_corridor_->GetRoadIds();
```

Comment:
why do we need to persist this information additionally in the agent?

Reply:
Maybe you can remember that we introduced this to support intersections to avoid wrong routing for intermediate vehicles. Road corridor is precalculated in the scenario generation and remembered by the agent.